clarity-ext-scripts/clarity_ext_scripts/covid/fetch_biobank_barcodes.py

```diff
@@ -128,3 +136,20 @@ def _plate_barcode_from(self, biobank_matrix):
 
     def _biobank_key(self, well, plate_barcode):
         return '{}_{}'.format(well, plate_barcode)
+
+    # TODO this has been duplicated from `validate_sample_creation_list`
+    #      fix this later.
+    def get_raw_sample_list(context):
+        file_name = "Raw sample list"
+        f = context.local_shared_file(file_name, mode="rb")
+
+        filtered = cStringIO.StringIO()
+        # Ignore everything at or after the line that contains this text:
+        stop_condition = "Sample Tracking Report Name"
+
+        for line in f:
+            if stop_condition in line:
+                break
+            filtered.write(line + "\n")
```

Comment:
I think there will be an empty line at the end. Unfortunately this has to be taken care of in the code. At line 100, you have to add something like "if len(line.values) == 1 continue".

Reply:
Not sure I follow here, since this has worked in the other case where it was used.